### ISSUE TITLE
Windows distribution via scoop

### ISSUE THREAD

**Issue by jonesbusy** [jenkins-infra/plugin-modernizer-tool#505]
### What feature do you want to see added?

The CLI is distributed by Homebrew which support Linux (and Windows users with WSL2) and MacOSX.

It would be great to not let Windows user left behind and allow to distribute the CLI via a windows package manager.

I woulds suggest scoop because it's very similar to Homebrew (a Tap is a Bucket, a Formala is an app manifest)

It also work with JRelease : https://jreleaser.org/guide/latest/reference/packagers/scoop.html

Good first issue if anyone is interested by packaging

**Comment by CodexRaunak** [comment]
Hey @jonesbusy I am interested in working on this one

**Comment by jonesbusy** [comment]
Sure feel free to open a PR

**Comment by gounthar** [comment]
What about Chocolatey?

**Comment by jonesbusy** [comment]
The point with Chocolatey is packages need to be packaged and pushed to https://push.chocolatey.org/ (A bit like maven central). So it means managing an API key

With scoop it's downloaded through anywhere (a bit like Homebrew) so we can reuse the uber jar already

What we miss is "just" creating the app manifest and updating the jrelease configuration

I think scoop is much simplier for now

**Comment by gounthar** [comment]
This issue was discussed during [the Infra Weekly Meeting](https://community.jenkins.io/tag/sig-infra).

In Jenkins, we currently utilize package managers such as Chocolatey and Homebrew, alongside the default Windows package manager.

If we decide to incorporate Scoop, we, as a community, must commit to keeping it updated and functioning properly. I am concerned that our infra team may not have the necessary bandwidth to manage this additional responsibility effectively. Should we successfully maintain Scoop, it could then be feasible to consider migrating the official Jenkins clients to a Scoop registry.

**Comment by jonesbusy** [comment]
Yes but here we are not talking about Jenkins for end user but a tool for plugin maintainer. I think it's a bit different.

Using scoop for one tool doesn't mean we should also use scoop to distribute Jenkins right ? 

For HomeBrew, Jenkins is not managed by the infra-team since it's packaged by a 3rd party (See https://www.jenkins.io/download/lts/macos/) right ? It's the same for scoop https://bjansen.github.io/scoop-apps/main/jenkins/. A 3rd party decided to package it and make it available (with a very outdated version by the way)

I will check the recording when published because I'm not sure to understand. In therm of maintenance it's fully automated by JReleaser : https://github.com/jenkins-infra/plugin-modernizer-tool/pull/515/files We are not even supposed to commit on the scoop git repository since it's not during the release process

**Comment by NawazTalha** [comment]
Hi, I’d like to work on this issue. Can you please assign it to me?
@jonesbusy

**Comment by jonesbusy** [comment]
Thanks for your interest but we don't assign issue.

Also I suggest to check the history of this issue to find there is already a PR open for it: https://github.com/jenkins-infra/plugin-modernizer-tool/pull/515

**Comment by gounthar** [comment]
And we'll have to wait a little, as the infra team is super busy with the aftermath of the Cloud Provider migration. 🤷
